File: backend/models/ActivityModel/Activity.cs

using FamilyPlanner.Models.FamilyModel;
using FamilyPlanner.Models.PlannedDayModel;
using FamilyPlanner.Models.PlannedTaskModel;
using FamilyPlanner.Models.UserModel;

namespace FamilyPlanner.Models.ActivityModel {
    public sealed class Activity {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool Completed { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string? UserUid { get; set; }
        public List<User>? Users { get; set; }
        public int? FamilyId { get; set; }
        public Family? Family { get; set; }
        public int? PlannedDayId { get; set; }
        public PlannedDay? PlannedDay { get; set; }
    }
}

File: familyPlanner/src/models/activity.ts
import { Family } from "./family";
import { PlannedDay } from "./plannedDay";
import { User } from "./user";

export interface Activity {
	id: number;
	name: string;
	description?: string;
	startDate: Date;
	endDate: Date;
	userUid?: string;
	users?: User[];
	familyId?: number;
	family?: Family;
	plannedDayId?: number;
	plannedDay?: PlannedDay;
	completed?: boolean;
}

export interface ActivityInput {
	startDate: Date;
	endDate: Date;
	name: string;
	description?: string;
	userUid?: string;
	users?: User[];
	planId?: number;
}

export interface ActivityUpdate {
	startDate: Date;
	endDate: Date;
	name: string;
	description?: string;
	userUid?: string;
	users?: User[];
	planId?: number;
}
